### PR TITLE
vulkan-cts: 1.3.7.0 -> 1.3.7.2

### PR DESCRIPTION
**Pull request by Flakebi**
## Description of changes

Changelog: https://github.com/KhronosGroup/VK-GL-CTS/releases/tag/vulkan-cts-1.3.7.1
Changelog: https://github.com/KhronosGroup/VK-GL-CTS/releases/tag/vulkan-cts-1.3.7.2
Builds and works.

Also add an impureTest that runs a test on the GPU.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by NickCao** [comment]
We do have lavapipe, the software vulkan implementation, is it possible to run cts on that?

**Comment by Flakebi** [comment]
Good idea. Seems to work just fine, I changed to a normal test using lavapipe.

File: pkgs/tools/graphics/vulkan-cts/default.nix [hunk 1]
```diff
@@ -34,82 +37,86 @@ let
...
-      --add-flags "--deqp-archive-dir=$out/archive-dir"
-  '';
+      wrapProgram $out/bin/deqp-vk \
+        --add-flags '--deqp-vk-library-path=${vulkan-loader}/lib/libvulkan.so' \
+        --add-flags "--deqp-archive-dir=$out/archive-dir"
+    '';
 
-  passthru.updateScript = ./update.sh;
+    passthru.updateScript = ./update.sh;
+    passthru.tests.run = runCommand "vulkan-cts-lavapipe-run" { nativeBuildInputs = [ pkg ]; } ''
```
Comment (NickCao) [inline pkgs/tools/graphics/vulkan-cts/default.nix:109]: You can use `finalAttrs.finalPackage` here.

File: pkgs/tools/graphics/vulkan-cts/default.nix [hunk 2]
```diff
@@ -1,6 +1,8 @@
 { lib, stdenv
+, callPackage
```
Comment (NickCao) [inline pkgs/tools/graphics/vulkan-cts/default.nix:2]: Unused input.

File: pkgs/tools/graphics/vulkan-cts/default.nix [hunk 3]
```diff
@@ -104,6 +107,10 @@ stdenv.mkDerivation (finalAttrs: {
   '';
 
   passthru.updateScript = ./update.sh;
+  passthru.tests.run = runCommand "vulkan-cts-lavapipe-run" { nativeBuildInputs = [ finalAttrs.finalPackage ]; } ''
```
Comment (NickCao) [inline pkgs/tools/graphics/vulkan-cts/default.nix:110]:
```suggestion
  passthru.tests.lavapipe = runCommand "vulkan-cts-tests-lavapipe" { nativeBuildInputs = [ finalAttrs.finalPackage ]; } ''
```

**Review by NickCao — COMMENTED** [review]
Some nitpicks, overall LGTM.

File: pkgs/tools/graphics/vulkan-cts/default.nix [hunk 4]
```diff
@@ -104,6 +106,10 @@ stdenv.mkDerivation (finalAttrs: {
   '';
 
   passthru.updateScript = ./update.sh;
+  passthru.tests.lavapipe = runCommand "vulkan-cts-tests-lavapipe" { nativeBuildInputs = [ finalAttrs.finalPackage ]; } ''
+    VK_ICD_FILENAMES=${mesa.drivers}/share/vulkan/icd.d/lvp_icd.x86_64.json deqp-vk -n dEQP-VK.api.smoke.triangle
```
Comment (NickCao) [inline pkgs/tools/graphics/vulkan-cts/default.nix:110]: Oh wait, is hardcoding 86_64 intended!

Comment (Flakebi) [reply pkgs/tools/graphics/vulkan-cts/default.nix]: Now using a `*` with some weird bash because `export` does not expand wildcards.

**Review by NickCao — APPROVED** [review]

**Pull request merged**